sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -359,13 +362,26 @@ def _get_snapshots(
             A dictionary of snapshot ids to snapshots for ones that could be found.
         """
         query = (
-            exp.select("snapshot")
+            exp.select(exp.to_column(f"{self.snapshots_table}.snapshot"))
             .from_(self.snapshots_table)
-            .where(None if snapshot_ids is None else self._snapshot_id_filter(snapshot_ids))
+            .where(
+                None
+                if snapshot_ids is None
+                else self._snapshot_id_filter(snapshot_ids, self.snapshots_table)
+            )
         )
         if hydrate_seeds:
-            query = query.select("content").join(
-                self.seeds_table, using=["name", "identifier"], join_type="left outer"
+            query = query.select(exp.to_column(f"{self.seeds_table}.content")).join(
+                self.seeds_table,
+                on=exp.and_(
+                    *[
+                        exp.to_column(f"{self.snapshots_table}.{col}").eq(
```

Comment:
would prefer exp.column() so we can avoid parsing

Reply:
`self.snapshots_table` is of the form `"schema.table"` (no quotes between the two), which doesn't align with the sqlglot-generated `"schema"."table"` and errors. `to_column` parses/generates them with the correct quoting in the context of the query.